data/src/main/java/com/linkedin/data/schema/compatibility/AnnotationCompatibilityChecker.java

```diff
@@ -0,0 +1,193 @@
+              newerCheckContextAndResolvedProperty = newerResolvedPropertiesMap.get(pathSpec);
+          Map<String, Object> newerResolvedProperties = newerCheckContextAndResolvedProperty.getValue();
+
+          // If olderResolvedProperties or newerResolvedProperties contains the same namespace as the provided SchemaAnnotationHandler,
+          // we do the annotation check
+          if (olderResolvedProperties.containsKey(annotationNamespace) || newerResolvedProperties.containsKey(
+              annotationNamespace))
+          {
+            AnnotationCompatibilityResult result =
+                handler.annotationCompatibilityCheck(olderResolvedProperties, newerResolvedProperties,
+                    olderCheckContextAndResolvedProperty.getKey(), newerCheckContextAndResolvedProperty.getKey());
+            if (!result.isCheckAnnotation())
+            {
+              _log.info("There is no implementation of annotationCompatibilityCheck in SchemaAnnotationHandler: "
+                  + handler.getClass().getName());
+              break;
+            }
+            results.add(result);
+          }
+          newerResolvedPropertiesMap.remove(pathSpec);
+        }
+        else
+        {
+          if (olderResolvedProperties.containsKey(annotationNamespace))
+          {
+            // olderResolvedPropertiesMap has a PathSpec which the newResolvedPropertiesMap does not have,
+            // it could be either newer schema removed annotation or the field is not existed, pass
+            // an empty newerResolvedPropertiesMap and empty newerAnnotationContext to the annotation check.
+            AnnotationCompatibilityResult result =
+                handler.annotationCompatibilityCheck(olderResolvedProperties, new HashMap<>(),
+                    olderCheckContextAndResolvedProperty.getKey(), new SchemaAnnotationHandler.AnnotationCheckContext());
+            if (!result.isCheckAnnotation())
+            {
+              _log.info("There is no implementation of annotationCompatibilityCheck in SchemaAnnotationHandler: "
+                  + handler.getClass().getName());
+              break;
+            }
+            results.add(result);
+          }
+        }
+      }
+      for (Map.Entry<PathSpec, Pair<SchemaAnnotationHandler.AnnotationCheckContext, Map<String, Object>>> e : newerResolvedPropertiesMap
+          .entrySet())
+      {
+        Pair<SchemaAnnotationHandler.AnnotationCheckContext, Map<String, Object>>
+            newerCheckContextAndResolvedProperty = newerResolvedPropertiesMap.get(e.getKey());
+        Map<String, Object> newerResolvedProperties = newerCheckContextAndResolvedProperty.getValue();
+        if (newerResolvedProperties.containsKey(annotationNamespace))
+        {
+          // newerResolvedPropertiesMap has a PathSpec which the olderResolvedPropertiesMap does not have,
+          // it could be either newer schema add new annotations or adding new field with annotations, pass
```

Comment:
either because newer schema aded new annotations or added new field with annotations

Reply:
Updated the logic here.